### PR TITLE
Async Crypto Endeavour (+benchmarks!)

### PR DESCRIPTION
**Pull request by dignifiedquire**
Removes all external crypto libraries, except for those used in `libp2p-crypto` and `multihashing`. This means it drops the usage of node-forge and relies only on webcrypto in the browser. 

Needs https://github.com/libp2p/js-libp2p-crypto/pull/10
### Size

|  | non-minified | minified |
| --- | --- | --- |
| before | `1.8M` | `999K` |
| after | `606K` | `382K` |
### Performance

|  | Node.js 6.6.0 | Browser (Chrome) |
| --- | --- | --- |
| before | createKey x 0.40 ops/sec | createKey x 0.31 ops/sec |
|  | send x 0.15 ops/sec | send x 0.11 ops/sec |
| after | createKey x 8.66 ops/sec | createKey x 1.64 ops/sec |
|  | send x 3.61 ops/sec | send x 0.55 ops/sec |

### CODE REVIEW

**Comment by dignifiedquire** [comment]
There seems to be an issue with go interop on the verification step currently. Investigating currently.

**Comment by dignifiedquire** [comment]
> There seems to be an issue with go interop on the verification step currently. Investigating currently.

All good, go interop restored in the latest version.

File: benchmarks/send.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+const pair = require('pull-pair/duplex')
+const PeerId = require('peer-id')
+const crypto = require('libp2p-crypto')
+
+const secio = require('../src')
+
+const suite = new Benchmark.Suite('secio')
+const ids = []
+
+suite.add('createKey', function (d) {
```
Comment (diasdavid) [inline benchmarks/send.js]: what is `d`?

Comment (dignifiedquire) [inline benchmarks/send.js]: see the other PR or https://benchmarkjs.com/docs

Comment (diasdavid) [inline benchmarks/send.js]: I mean, can it have a better and intuitive name?

File: src/etm.js [hunk 2]
```diff
@@ -1,73 +1,70 @@
...
+
+        mac.digest(data, (err, digest) => {
+          if (err) {
+            return cb(err)
+          }
+
+          cb(null, Buffer.concat([
+            data,
+            digest
+          ]))
```
Comment (diasdavid) [inline src/etm.js]: this could be one line, it is just a `Buffer.concat`

**Comment by dignifiedquire** [comment]
@diasdavid should be ready

**Pull request merged**